learn/app-development/widgets/prefab/entity-extraction-from-documents.md

```diff
@@ -0,0 +1,34 @@
+---
+title: "Entity Extraction Prefab - Automatically extract data from structured documents such as Invoices, Receipts"
+id: ""
+---
+---
+
+There are many situations where data has to be extracted from a scanned recipt or a PDF document or an screenshot taken from a mobile phone. Automatically extracting information from such documents will enable in building delighful user experiences. This prefab will help implement such extraction of data any structured document. Given the document as input, the prefab uploads to it to [AWS Textract](https://docs.aws.amazon.com/textract/latest/dg/what-is.html) to extract data from it. The extracted data is available as key-value pair and developer can pick the data they are interested in.
+
+
+## How to use this prefab:
+
+1. Download the prefab [here](https://github.com/wavemaker/prefab-entity-extractor/releases/latest) and [import into your WaveMaker project](/learn/app-development/widgets/custom-widgets/#importing-prefabs).
+
+2. Create AWS account and IAM user with policy `AmazonTextractFullAccess`.
+
+3. Generate and set `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and the `AWS_REGION` at prefab’s App Environment. (In your app’s `Settings > Config profile > EntityEXtraction (prefab) > App Environment`)
+
+![/learn/assets/entity-extraction-from-document-picture1.png](/learn/assets/entity-extraction-from-document-picture1.png)
+
+4. Bind the “URL” and “Entities” (properties of prefab) on your page. URL is the path to the file (uploaded to the project) to be processed and Entities are labels from the document which are required to be extract their values.
```

Comment:
@ramicetty It is not clear how i get the url of the file to be given as input. can you explain more?

Reply:
@prasanth-reddy-wavemaker-com  it comes from the target App. For example, the document path can be stored in a database or it can be the value of another widget in an app. A developer can bind one of these scenarios to prefab's input-bound (URL).

Reply:
@prasanth-reddy-wavemaker-com but specified document must available to download for process.

Reply:
Can i bind it to output of fileupload?